### ISSUE TITLE
Rename output file on workflow

### ISSUE THREAD

**Issue by methylome** [galaxyproject/galaxy#1686]
I've tried to use this function for the output of tophat V0.7 for the input of a list of paired files, i.e, list:paired. First, #{input} doesn't work although it's a parameter name as shown below.
`<when value="paired_collection">
                <param format="fastqsanger" name="input" type="data_collection" collection_type="paired" label="RNA-Seq FASTQ paired reads" help="Must have Sanger-scaled quality values with ASCII offset 33" />
                <expand macro="paired_parameters" />
            </when>
`
Second, #{input1}, which means the first file name, works but all paired datasets have the same output name. For instance, given a list of two paired files: hgrna1 and hgrna2, each of which has two fastq files, two output bam file names are the same as hgrna1_R1.bam.

Is this a known bug?

Thanks,
Justin.

**Comment by jmchilton** [comment]
Is this the same problem as https://github.com/galaxyproject/galaxy/issues/1675?
- We need a more descriptive UI to show both element identifier and name for collection items.
- And we need #1680 - to be able to rename collection names on mapped steps of a workflow.

**Comment by dannon** [comment]
@jmchilton yes, I think so.   Per conversation in IRC yesterday I started taking a look here at the various issues with the rename action and am hoping to get a fix out today.

**Comment by chambm** [comment]
@dannon Is this supposed to be fixed? I have this problem with HISAT. See this simple test workflow at:
http://129.114.18.155/u/chambm/w/hisat-rename-test

This is a fresh 17.01 Galaxy hosted on Jetstream.

**Comment by dannon** [comment]
@chambm Let me doublecheck, but yeah, I thought this was resolved.

('close and comment' button fail, sorry)

**Comment by chambm** [comment]
Did you have a chance to check?

**Comment by dannon** [comment]
Closing — RenameDatasetAction is fully supported in the workflow editor's FormOutput.vue, including workflow variable substitution.
